step_exec_lib/utils/processes.py

```diff
@@ -16,25 +16,25 @@ def run_and_log(args: List[str], **kwargs: Any) -> subprocess.CompletedProcess:
     return run_res
 
 
-def run_and_handle_error(args: List[str], expected_error_text: str, **kwargs: Any) -> int:
+def run_and_handle_error(args: List[str], expected_error_text: str, **kwargs: Any) -> subprocess.CompletedProcess:
     logger.info("Running command:")
     logger.info(" ".join(args))
     if "text" not in kwargs:
         kwargs["text"] = True
 
-    try:
-        run_res = subprocess.run(args, **kwargs, check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)  # nosec
+    run_res = subprocess.run(args, **kwargs, check=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)  # nosec
```

Comment:
check=False allows us to control when we receive the error.

> If check is true, and the process exits with a non-zero exit code, a CalledProcessError exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.

https://docs.python.org/3/library/subprocess.html